kratos/utilities/pointer_communicator.h

```diff
@@ -78,23 +78,51 @@ class ResultsProxy
     /// Destructor.
     virtual ~ResultsProxy() {}
 
-    /**this function returns the effect of "user_function(gp)" both if the gp is locally owned
+    /**this function returns the effect of "user_function(rGlobalPointer)" both if the rGlobalPointer is locally owned
      * and if it is remotely owned
      */
-    TSendType Get(GlobalPointer<TPointerDataType>& gp)
+    TSendType Get(GlobalPointer<TPointerDataType>& rGlobalPointer) const
     {
-        if(gp.GetRank() == mCurrentRank)
-            return mUserFunctor(gp);
-        else
-            return mNonLocalData[gp];
+        if(rGlobalPointer.GetRank() == mCurrentRank)
+            return mUserFunctor(rGlobalPointer);
+        else {
+            auto non_local_gp = mNonLocalData.find(rGlobalPointer);
+#ifdef KRATOS_DEBUG
+            if(non_local_gp == mNonLocalData.end())
+                KRATOS_ERROR << "Missing entry in NonLocalData" << std::endl;
+#endif
+            return non_local_gp->second;
+        }
+    }
+
+    TSendType Get(const GlobalPointer<TPointerDataType>& rGlobalPointer) const
+    {
+        if(rGlobalPointer.GetRank() == mCurrentRank)
+            return mUserFunctor(rGlobalPointer);
+        else {
+            auto non_local_gp = mNonLocalData.find(rGlobalPointer);
+#ifdef KRATOS_DEBUG
+            if(non_local_gp == mNonLocalData.end())
+                KRATOS_ERROR << "Missing entry in NonLocalData" << std::endl;
+#endif
```

Comment:
```suggestion
            KRATOS_DEBUG_ERROR_IF(non_local_gp == mNonLocalData.end()) << "Missing entry in NonLocalData" << std::endl;
```

Reply:
Oh, yep I could have done that  :man_facepalming: